internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -2758,12 +2810,45 @@ func (r *Reconciler) reconcileRepos(ctx context.Context,
 	}
 
 	postgresCluster.Status.PGBackRest.Repos =
-		getRepoVolumeStatus(postgresCluster.Status.PGBackRest.Repos, repoVols, extConfigHashes,
-			replicaCreateRepo.Name)
+		getRepoVolumeStatus(postgresCluster.Status.PGBackRest.Repos, repoVols,
+			extConfigHashes, replicaCreateRepo.Name)
 
 	return replicaCreateRepo, utilerrors.NewAggregate(errors)
 }
 
+// +kubebuilder:rbac:groups="",resources="pods",verbs={list}
+
+// getRepoHostVolumeRequests gets the pgBackRest repo host volume request annotations
+// from the repo host Pod and stores them in the Postgres Cluster object's status.
+// If there is no repo host or there are no annotations, this function returns
+// without updating the status.
+func (r *Reconciler) getRepoHostVolumeRequests(ctx context.Context,
+	cluster *v1beta1.PostgresCluster) error {
+
+	pods := &corev1.PodList{}
+	err := errors.WithStack(
+		r.Client.List(ctx, pods,
+			client.InNamespace(cluster.Namespace),
+			client.MatchingLabelsSelector{
+				Selector: naming.PGBackRestDedicatedLabels(cluster.Name).AsSelector()},
+		))
+
+	if len(pods.Items) > 0 {
```

Comment:
If we always expect 1 pod should the check be this ❓ 
```suggestion
	if len(pods.Items) == 1 {
```

Reply:
That's a fair point. I was initially thinking it would allow it to be a bit more robust, but if there are two repo hosts for some reason, we probably don't want to do anything until the next reconcile anyway. I'll make that change.